core/trino-main/src/main/java/io/trino/sql/gen/columnar/ColumnFilterProcessor.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.gen.columnar;
+
+import io.trino.operator.project.SelectedPositions;
+import io.trino.spi.Page;
+
+import static io.trino.operator.project.SelectedPositions.positionsList;
+import static io.trino.operator.project.SelectedPositions.positionsRange;
+
+public class ColumnFilterProcessor
```

Comment:
Actually after reading the code for the users of this class. `between`, `call`, `isNotNull`, `isNull`, and `in`. I'm not sure why the these 5 wrapper classes exist at all.  The seem to be a few lines of wrapper code.  Are they there simply to help with debugging heap dumps or such?

If they aren't needed, I'd make this class implement `ExpressionEvaluator` and use this class directly instead of the 4 wrappers.  I'd move the factory methods in the wrappers to where ever `createColumnarFilterEvaluator` ends up.

Reply:
I got rid of those 5 wrapper classes